component/module/file/file.go

```diff
@@ -69,24 +72,43 @@ var (
 // New creates a new module.file component.
 func New(o component.Options, args Arguments) (*Component, error) {
 	c := &Component{
+		opts: o,
 		mod:  module.NewModuleComponent(o),
 		args: args,
 	}
+	defer c.isCreated.Store(true)
 
 	var err error
 	c.managedLocalFile, err = c.NewManagedLocalComponent(o)
 	if err != nil {
 		return nil, err
 	}
-
 	if err := c.Update(args); err != nil {
 		return nil, err
 	}
 	return c, nil
 }
 
+// NewManagedLocalComponent creates the new local.file managed component.
+func (c *Component) NewManagedLocalComponent(o component.Options) (*file.Component, error) {
```

Comment:
Does this need to be exported as part of the public API of the component? Seems like we can make it internal, right?

Reply:
good catch, will fix tomorrow